### PR TITLE
Provision for user config override for name-service and some utils code refactoring

### PR DESCRIPTION
**Pull request by SanchayPahalwani**

### CODE REVIEW

File: src/packages/gaiaservice.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        const payload = {
+            claim: content,
+            issuer: { publicKey },
+            subject: { publicKey },
+        };
+        const token = tokenSigner.sign(payload);
+        return [blockstack.wrapProfileToken(token)];
+    }
+
+    private _sanitizePrivKey = (privKey: string): string => {
```
Comment (nandubatchu) [inline src/packages/gaiaservice.ts]: - [ ] this method can be made static I suppose, and moved to utils file.

File: src/index.ts [hunk 2]
```diff
@@ -158,32 +159,37 @@ class CruxPayPeer extends EventEmitter {
...
-            this._nameservice = await configService.getBlockstackServiceForConfig();
-        }
+        let configService;
 
         if (this._hasPayIDClaimStored()) {
             const payIDClaim = this._storage.getJSON("payIDClaim");
             // log.debug(`Local payIDClaim:`, payIDClaim)
             this._setPayIDClaim(new PayIDClaim(payIDClaim as ICruxPayClaim, { getEncryptionKey: this._getEncryptionKey }));
+            const ns: blockstackService.BlockstackService = new blockstackService.BlockstackService();
+            const status = await ns.getRegistrationStatus({secrets: (this._payIDClaim as PayIDClaim).identitySecrets});
```
Comment (nandubatchu) [inline src/index.ts]: status will always return NONE unless identity is restored with the method `restoreIdentity` which will set the private property `_identityCouple`.

File: src/packages/gaia-service/utils.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        gaiaRead = await getGaiaReadUrl(gaiaHub as string);
+    }
+    return {
+        gaiaReadUrl: gaiaRead,
+        gaiaWriteUrl: gaiaHub,
+        ownerAddress: bitcoinAddress,
+    };
+};
+
+export const getGaiaReadUrl = async (gaiaHubURL: string): Promise<string> => {
```
Comment (quixote911) [inline src/packages/gaia-service/utils.ts]: Please rename `gaiaHubURL` to specify writeUrl or readUrl. Lets not use just 'gaiaHubURL' anywhere.

File: src/packages/gaia-service/utils.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        if (addressFromPub === gaiaDetails.ownerAddress) {
+            finalContent = content;
+        } else {
+            throw ErrorHelper.getPackageError(PackageErrorCode.CouldNotValidateZoneFile);
+        }
+    }
+    return finalContent;
+};
+
+export let getGaiaHubUrlsFromBlockstackID = async (blockstackId: string) => {
```
Comment (quixote911) [inline src/packages/gaia-service/utils.ts]: Rename this function to - `getGaiaDataFromBlockstackId` and define the output of this function as an interface `GaiaData`

File: src/index.ts [hunk 5]
```diff
@@ -158,32 +159,40 @@ class CruxPayPeer extends EventEmitter {
...
 
         if (this._hasPayIDClaimStored()) {
             const payIDClaim = this._storage.getJSON("payIDClaim");
             // log.debug(`Local payIDClaim:`, payIDClaim)
             this._setPayIDClaim(new PayIDClaim(payIDClaim as ICruxPayClaim, { getEncryptionKey: this._getEncryptionKey }));
+            const ns: blockstackService.BlockstackService = new blockstackService.BlockstackService();
+            await this.getPayIDClaim().decrypt();
+            await ns.restoreIdentity(this.getPayIDClaim().virtualAddress as string, {secrets: (this._payIDClaim as PayIDClaim).identitySecrets});
+            const status = await ns.getRegistrationStatus({secrets: this.getPayIDClaim().identitySecrets});
+            await this.getPayIDClaim().encrypt();
```
Comment (nandubatchu) [inline src/index.ts]: For internal usage, please rely on `this._payIDClaim` rather than `this.getPayIDClaim()`

**Pull request merged**